### PR TITLE
Remove AWS Pytorch channel in examples

### PR DESCRIPTION
**Pull request by junpuf**
*Issue #, if available:*
#444 

*Description of changes:*
The AWS PyTorch Conda channel is being deprecated, future development will be stopped, so removing the usage of it. 

---
### Testing
Test infra is 2 p4d using pcluster with fsx and slurm built using DLAMI (the same AMI used by HyperPod AMI).
#### 10.FSDP
- Passed test
- Below environment variables are no longer required in newer aws-ofi-nccl versions. 
```
export FI_EFA_USE_DEVICE_RDMA=1 # use for p4d
export FI_EFA_FORK_SAFE=1
```
- Fixed README.md

#### 16.pytorch-cpu-ddp
- Passed test
- updated conda env creation process

#### 17.SM-modelparallelv2
For 17.SM-modelparallelv2, it seems `pytorch="2.2.0=sm_py3.10_cuda12.1_cudnn8.9.5_nccl_pt_2.2_tsm_2.3_cuda12.1_0` declared dependency on `aws-ofi-nccl >=1.7.1,<2.0` (probably due to copying the build recipe from aws conda channel). Because of this, i made a workaround by supplying the 2 binaries needed for this pytorch package.

```
Could not solve for environment specs
The following package could not be installed
└─ pytorch ==2.2.0 sm_py3.10_cuda12.1_cudnn8.9.5_nccl_pt_2.2_tsm_2.3_cuda12.1_0 is not installable because it requires
   └─ aws-ofi-nccl >=1.7.1,<2.0 , which does not exist (perhaps a missing channel).
```

Workaround included 2 binaries (details below) required in a new `bin` directory inside the example directory.

```
# dependency package for pytorch ==2.2.0 sm_py3.10_cuda12.1_cudnn8.9.5_nccl_pt_2.2_tsm_2.3_cuda12.1_0
aws-ofi-nccl-1.7.4-aws_0.tar.bz2 
# dependency package for aws-ofi-nccl-1.7.4-aws_0.tar.bz2 
hwloc-2.9.2-h2bc3f7f_0.tar.bz2
```
#### 20.FSDP-Mamba
Simply removed commented lines that referenced the AWS PyTorch conda channel, no test necessary

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: 3.test_cases/10.FSDP/1.distributed-training.sbatch [hunk 1]
```diff
@@ -20,11 +20,7 @@ GPUS_PER_NODE=8 # 4 for G5.12x, 8 for P4/P5
 ###########################
 
 ## Plenty of EFA level variables
-## Comment out for non-efa instances (G4d, P3)
-## For G5.12x, Comment out RDMA and Fork safe
 ## For G4dn and other G5, comment out all
-export FI_EFA_USE_DEVICE_RDMA=1 # use for p4d
-export FI_EFA_FORK_SAFE=1
 export FI_LOG_LEVEL=1
```
Comment (mhuguesaws) [inline 3.test_cases/10.FSDP/1.distributed-training.sbatch:24]:
```suggestion
export FI_LOG_LEVEL=warn
```

File: 3.test_cases/10.FSDP/README.md [hunk 2]
```diff
@@ -36,24 +36,12 @@ If you'd like to instead use your own dataset, you can do so by [formatting it a
 
 ## 3. Launch Training
 
-- The script to launch a Llama 2 Slurm batch training job can be found in `1.distributed_training.sbatch`.
-- The script to launch a Mixtral training can be found in `2.distributed_training_mixtral.sbatch`.
+- The script to launch a Llama 2 Slurm batch training job can be found in `1.distributed-training.sbatch`.
```
Comment (mhuguesaws) [inline 3.test_cases/10.FSDP/README.md:39]: Why not calling it `distributed-training-llama2.sbatch` ?

**Review by mhuguesaws — COMMENTED** [review]
Lef comments

**Comment by sean-smith** [comment]
@junpuf why is this marked as a draft? can we mark it ready for review

**Comment by junpuf** [comment]
> @junpuf why is this marked as a draft? can we mark it ready for review

Finishing up last bit of testing, then i'll mark it ready for review.

**Comment by junpuf** [comment]
For 17.SM-modelparallelv2, it seems `pytorch="2.2.0=sm_py3.10_cuda12.1_cudnn8.9.5_nccl_pt_2.2_tsm_2.3_cuda12.1_0` declared dependency on `aws-ofi-nccl >=1.7.1,<2.0` (probably due to copying the build recipe from aws conda channel). I'm figuring out an workaround at the moment.

**Review by sean-smith — APPROVED** [review]
LGTM. In the future let's remove the `aws-ofi-nccl` and `hwloc` dependencies.

**Pull request merged**